src/lib/io/sol-iio.c

```diff
@@ -179,7 +184,11 @@ check_trigger_name(const char *trigger_dir, const char *trigger_name)
         return false;
     }
 
-    result = streq(trigger_name, name);
+    /* For hrtimer trigger, the current trigger name is the string after HRTIMER_TRIGGER_PREFIX */
+    if (streqn(trigger_name, HRTIMER_TRIGGER_PREFIX, HRTIMER_TRIGGER_PREFIX_LEN))
```

Comment:
There is a `strstartswith` on sol-util-internal.h. Maybe it could be useful here?